### PR TITLE
[VMware/NGC] Add Other Storage Interfaces like Mount/Unmount, Snapshot Volumes 

### LINKED ISSUE #273: [VMware/NGC] VMware vSphere OpenSDS Storage Plugin Add Other Interfaces like Mount/Unmount, Snapshot Volumes
**Is this a BUG REPORT or FEATURE REQUEST?**:
 /kind feature

**What you expected to happen**:
Add Other Storage Interfaces for Mount/Unmount, Snapshot Volumes etc.

### PR DESCRIPTION
**Pull request by himanshuvar**
**What this PR does / why we need it**:
With reference to opensds/design-specs#38, This PR is to
support VMware vSphere web client plugin  for OpenSDS storage with other storage Interfaces like Mount/Unmount , Snapshot Volumes etc.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #273 

**Special notes for your reviewer**: 
This PR includes
NGC Plugin, Adapter Manager Changes and Some updation for Registration messages.

known limitations:
1. No Restore/Rollback Snapshot.

**Testing notes**:
1. Tested unmounted i.e available volume listing.
2. Tested Volume Snapshot.
3. Tested snapshots listing.
4. Tested Volume Mount to the ESXI host.

### CODE REVIEW

File: vmware/ngc/NGC-Plugin/html/src/main/webapp/resources/html/host/host-left.html [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+			$("#target_img").css("display", "none");
+			$("#lun_img").css("display", "block");
+			$("#fs_img").css("display", "none");
+			$("#topn_img").css("display", "none");
+			$("#test_img").css("display", "none");
+			parent.$("#mainFrame").attr("src", "luns.html");
+		} /*else if (type == "targets") {
+			$('#herfDatastore').css('color', '#0068FF');
+			$('#hrefLun').css('color', '#0068FF');
+			$('#hrefTargets').css('color', '#000');
```
Comment (wisererik) [inline vmware/ngc/NGC-Plugin/html/src/main/webapp/resources/html/host/host-left.html]: please remove unused code

File: vmware/ngc/NGC-Plugin/html/src/main/webapp/resources/html/vm/vm-left.html [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+			$('#hrefVirtualDisk').css('color', '#0068FF');
+			$('#hrefRawDevice').css('color', '#0068FF');
+			$('#hrefLuns').css('color', '#000');
+
+			$("#luns_img").css("display", "block");
+			$("#virtualDisk_img").css("display", "none");
+			$("#rawDevice_img").css("display", "none");
+			parent.$("#mainFrame").attr("src", "vmLuns.html");
+		}*/
+	}
```
Comment (wisererik) [inline vmware/ngc/NGC-Plugin/html/src/main/webapp/resources/html/vm/vm-left.html]: same here

File: vmware/ngc/NGC-Plugin/html/src/main/webapp/resources/js/snapshot/snapshotTab.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+	//行点击事件
+	/*$("#snapshotTable tbody tr").bind("click", function () {
+
+	    if ($(this).find("td").eq(2).find("input").is(":checked")) {
+	        $(this).find("td").eq(2).find("input").prop("checked", false);
+	    } else {
+	        $(this).find("td").eq(2).find("input").prop("checked", true);
+	    }
+
+	    clickSingleCheckBox();
```
Comment (wisererik) [inline vmware/ngc/NGC-Plugin/html/src/main/webapp/resources/js/snapshot/snapshotTab.js]: same here

File: vmware/ngc/NGC-Plugin/html/src/main/webapp/resources/html/host/host-left.html (hunk 1 above)
Comment (himanshuvar) [reply vmware/ngc/NGC-Plugin/html/src/main/webapp/resources/html/host/host-left.html]: Done

File: vmware/ngc/NGC-Plugin/html/src/main/webapp/resources/html/vm/vm-left.html (hunk 2 above)
Comment (himanshuvar) [reply vmware/ngc/NGC-Plugin/html/src/main/webapp/resources/html/vm/vm-left.html]: Done

File: vmware/ngc/NGC-Plugin/html/src/main/webapp/resources/js/snapshot/snapshotTab.js (hunk 3 above)
Comment (himanshuvar) [reply vmware/ngc/NGC-Plugin/html/src/main/webapp/resources/js/snapshot/snapshotTab.js]: Done

File: vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/RestClient.java [hunk 4]
```diff
@@ -275,4 +278,63 @@ JSONObject detachVolume(String volumeId) throws Exception {
         }
         return responseDelete;
     }
+
+    JSONObject getVolume(String volumeId) throws Exception {
+
+        JSONArray response = (JSONArray)request.get(String.format("/block/volumes?wwn=%s", volumeId));
```
Comment (NajmudheenCT) [inline vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/adapters/opensds/RestClient.java]: Based on the Unique identifier implemntaion in hotpot , we have to chage this filter (?wwn) . 
issue raised https://github.com/opensds/nbp/issues/286

File: vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/common/Storage.java [hunk 5]
```diff
@@ -19,21 +19,90 @@
...
     public abstract StoragePoolMO getStoragePool(String poolId) throws Exception;
+
     public abstract void attachVolume(String volumeId, ConnectMO connect) throws Exception;
+
     public abstract void detachVolume(String volumeId, ConnectMO connect) throws Exception;
+
+    /**
+     * query the volume mob by volume id
+     *
+     * @param volumeId in oceanstor volume id is wwn
```
Comment (skdwriting) [inline vmware/ngc/NGC-AdapterManager/src/main/java/org/opensds/vmware/ngc/common/Storage.java]: Is it specific to OceanStor, do we mention it here?

**Pull request merged**